### PR TITLE
ArgumentParser use for iOS, tvOS, watchOS to help with UITesting

### PR DESCRIPTION
**Pull request by thecb4**
UITesting allows you to pass arguments, but you still have to parse arguments inside your app. I would use SPM's ArgumentParser utility to do this. Currently iOS, watchOS, tvOS are not supported. The changes are fairly straight forward.

Add @available, #available flags for iOS, watchOS, tvOS in TSCBasic

### CODE REVIEW

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by thecb4** [comment]
@ddunbar Something I noticed. I think this may have been mentioned somewhere else. 
This compiles and runs, but causes the SwiftUI canvas to error out due to Xcode attempting to recompile the whole library instead of just the dependency. I pulled the Argument Parser code into a separate package and SwiftUI previews work.

https://github.com/thecb4/swift-argument-parser

Steps to recreate:
1. create new SwiftUI project and include https://github.com/thecb4/swift-package-manager as dependency
2. Select ContentView. SwiftUI canvas errors out on SwiftPM compile (although not a dependency of the project)
3. Create new SwiftUI project and include https://github.com/thecb4/swift-argument-parser
4. Select ContentView. SwiftUI canvas shows correctly

**Comment by ddunbar** [comment]
@thecb4 would you be willing to put that info in a bug for Apple?

File: Package.swift [hunk 1]
```diff
@@ -14,6 +14,9 @@ import PackageDescription
 
 let package = Package(
     name: "SwiftPM",
+    platforms: [
+        .macOS(.v10_10), .iOS(.v9),
```
Comment (aciidb0mb3r) [inline Package.swift]: I don't think we need to specify these.

**Comment by thecb4** [comment]
I’ll remove and show the errors in XCode.

It Referenced that need for iOS 9, but the package was supporting 8 because nothing was specified. The error went away by introducing.

**Comment by thecb4** [comment]
I'll make all the changes

File: Package.swift (hunk 1 above)
Comment (aciidb0mb3r) [reply Package.swift]: Rest of the changes look good! Can you remove this section since we're using availability for newer APIs.

Comment (thecb4) [reply Package.swift]: Should be removed now

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@thecb4 Thanks!

**Comment by thecb4** [comment]
Awesome!! Thank you!

**Pull request merged**